### PR TITLE
chore: Restore self-deleting messages SQSERVICES-1059

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQSERVICES-1059" title="SQSERVICES-1059" target="_blank">TaskSQSERVICES-1059</a>  Restore self-deleting messages
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Restore self-deleting messages

Needs releases with:

- [ wireapp/wire-ios-data-model/pull/1293 ] 
- [ wireapp/wire-ios-request-strategy/pull/432 ]

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by johnxnguyen — APPROVED** [review]

File: Source/Synchronization/ZMHotFix.m [hunk 1]
```diff
@@ -61,7 +61,8 @@ - (void)applyPatches
         return;
     }
     
-    NSString * currentVersionString = [[[NSBundle bundleForClass:ZMUserSession.class] infoDictionary] objectForKey:@"CFBundleShortVersionString"];
+   /// Normally we fetch the short version from the framework bundle, but it's not working so we hardcode the version.
+    NSString * currentVersionString = @"402";
```
Comment (typfel) [inline Source/Synchronization/ZMHotFix.m:65]: 402.0.0?

**Review by typfel — CHANGES_REQUESTED** [review]
Copy the full version from version.xconfig?

File: Source/Synchronization/ZMHotFix.m (hunk 1 above)
Comment (KaterinaWire) [reply Source/Synchronization/ZMHotFix.m]: Yes, fixed.
`402.0.0` will be the new version.

**Comment by KaterinaWire** [comment]
> Copy the full version from version.xconfig?

I've changed the version in `version.xconfig`

**Review by typfel — APPROVED** [review]

**Comment by KaterinaWire** [comment]
release major

**Comment by KaterinaWire** [comment]
release major

**Comment by billypchan** [comment]
release major

**Comment by billypchan** [comment]
release major

**Comment by billypchan** [comment]
release major

**Pull request merged**